### PR TITLE
[BugFix] Correct DreamerV3 symexp two-hot semantics

### PR DESCRIPTION
**Pull request by vmoens**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.17.0) (oldest at bottom):
* #4095
* #4075
* #4074
* #4073
* #4072
* #4068
* #4067
* #4066
* __->__ #4065

Summary:
- add the exported SymExpTwoHot representation and shared encode, decode, and cross-entropy helpers
- construct exact symmetric raw supports from symexp(linspace(-20, 20, bins))
- train reward and value logits against raw-space interpolated targets
- document the public API and cover official numerical fixtures, saturation, symmetry, state, and torch.compile

Rationale:
DreamerV3 categorical scalar predictions operate on raw symexp support values. The previous implementation interpolated symlog targets over a linear support, which changed both the training targets and decoded magnitudes. A single representation now defines and tests the complete contract.

Test plan:
- uv run pytest -q test/objectives/test_dreamer_v3.py
- ufmt check torchrl/modules/models/model_based_v3.py torchrl/objectives/dreamer_v3.py test/objectives/test_dreamer_v3.py
- git diff --check

### CODE REVIEW

**Comment by vmoens** [comment]
**Update after the stack-level ghstack rebase:** The full eight-PR Dreamer stack was checked out from #4075, rebased onto current `origin/main`, linted with `pre-commit run --all-files`, and submitted successfully as `vmoens`. This PR is now at `487a458d101e`. The core Dreamer suites pass (267 tests) and the two doctest modules pass (15 tests); the DMC-only smoke still requires `dm_control`, which is not installed locally. Any earlier conflict/head-status note below is superseded; the substantive code-review observations remain applicable.

Reviewed the freshly rebased head `071e813a862e`, including the complete layer diff, the current DreamerV3 reference implementation, and the latest PR discussion. The raw `symexp(linspace)` support, raw-space interpolation, and symmetric paired decode are internally consistent and match the current reference structure.

One actionable test issue remains at `test/objectives/test_dreamer_v3.py:366`: the new compile-parity assertion is flaky. A focused run passed, but the subsequent full file run failed with an eager/compiled difference of 8.0 on an approximately 1.9e6 output (relative error 4.13e-6), exceeding the default `assert_close` tolerance. Because `logits` is unconstrained random data and the five-bin support reaches approximately 4.85e8, this can depend on the sample and reduction lowering. Please use a fixed moderate logits fixture and/or an explicit float32-appropriate tolerance so this test is stable while still checking compile parity.

Checks: the four new SymExpTwoHot-focused tests passed in isolation; the full `test/objectives/test_dreamer_v3.py` run was 44 passed / 1 failed at the assertion above; `git diff --check` is clean. CI has restarted for the rebased head.

**Comment by vmoens** [comment]
Addressed the review feedback and resubmitted the rebased stack. I made the compile-parity test deterministic with fixed moderate logits and explicit float32 tolerances, and preserved the lint-safe public re-exports. The focused compile test and the full all-files lint suite pass at this PR boundary.
